gcode.c

```diff
@@ -1060,8 +1100,29 @@ status_code_t gc_execute_block (char *block)
             }
 
             continue;
-        }
+#if STRING_REGISTERS_ENABLE
+        } else if (letter == '@') {
+            if(gc_state.skip_blocks)
+                return Status_OK;
+            float register_id;
+            if (!read_float(block, &char_counter, &register_id)) {
```

Comment:
ah, this limits the setting of a string register to exact values. Should use read_param here 🤔

Reply:
Fixed